model/RdsResultStorage.php

```diff
@@ -189,81 +136,48 @@ public function storeItemVariables(
                 $callIdItem
             );
         }
-        
+
         $this->getPersistence()->insertMultiple(self::VARIABLES_TABLENAME, $dataToInsert);
     }
 
-    /*
-     * retrieve specific parameters from the resultserver to configure the storage
-     */
-    public function configure($callOptions = array())
+    public function storeRelatedTestTaker($deliveryResultIdentifier, $testTakerIdentifier)
     {
-        \common_Logger::d('configure  RdsResultStorage with options : ' . implode(" ", $callOptions));
+        $this->storeRelatedData($deliveryResultIdentifier, self::TEST_TAKER_COLUMN, $testTakerIdentifier);
     }
 
-    /**
-     * Store test-taker doing the test
-     * @param $deliveryResultIdentifier
-     * @param $testTakerIdentifier
-     */
-    public function storeRelatedTestTaker($deliveryResultIdentifier, $testTakerIdentifier)
+    public function storeRelatedDelivery($deliveryResultIdentifier, $deliveryIdentifier)
     {
-        $sql = 'SELECT COUNT(*) FROM ' . self::RESULTS_TABLENAME .
-            ' WHERE ' . self::RESULTS_TABLE_ID . ' = ?';
-        $params = array($deliveryResultIdentifier);
-        if ($this->getPersistence()->query($sql, $params)->fetchColumn() == 0) {
-            $this->getPersistence()->insert(
-                self::RESULTS_TABLENAME,
-                array(
-                    self::TEST_TAKER_COLUMN => $testTakerIdentifier,
-                    self::RESULTS_TABLE_ID => $deliveryResultIdentifier
-                )
-            );
-        } else {
-            $sqlUpdate = 'UPDATE ' . self::RESULTS_TABLENAME . ' SET ' . self::TEST_TAKER_COLUMN . ' = ? WHERE ' . self::RESULTS_TABLE_ID . ' = ?';
-            $paramsUpdate = array($testTakerIdentifier, $deliveryResultIdentifier);
-            $this->getPersistence()->exec($sqlUpdate, $paramsUpdate);
-        }
+        $this->storeRelatedData($deliveryResultIdentifier, self::DELIVERY_COLUMN, $deliveryIdentifier);
     }
 
     /**
      * Store Delivery corresponding to the current test
-     * @param $deliveryResultIdentifier
-     * @param $deliveryIdentifier
+     * @param string $deliveryResultIdentifier
+     * @param string $relatedField
+     * @param string $relatedIdentifier
      */
-    public function storeRelatedDelivery($deliveryResultIdentifier, $deliveryIdentifier)
+    public function storeRelatedData($deliveryResultIdentifier, $relatedField, $relatedIdentifier)
     {
-        $sql = 'SELECT COUNT(*) FROM ' . self::RESULTS_TABLENAME .
-            ' WHERE ' . self::RESULTS_TABLE_ID . ' = ?';
-        $params = array($deliveryResultIdentifier);
-        if ($this->getPersistence()->query($sql, $params)->fetchColumn() == 0) {
+        $qb = $this->getQueryBuilder()
+            ->select('COUNT(*)')
+            ->from(self::RESULTS_TABLENAME)
+            ->andWhere(self::RESULTS_TABLE_ID .' = :id')
+            ->setParameter('id', $deliveryResultIdentifier);
+        if ($qb->execute()->fetchColumn() == 0) {
```

Comment:
`===`

Reply:
In fact, the result is returned as a string. I could test it this way or with `=== '0'`.
Maybe `(int) $qb->execute()->fetchColumn() === 0` is better.
What do you think?

Reply:
It's always more safe if you check the type as well. Just think about `null == 0 vs null === 0`, etc... Personally I would prefer the 2nd version.

Reply:
Ok, fixed!